scripts/manage-db.py

```diff
@@ -31,40 +31,75 @@ def cleanup_releases(trans, nightly_age, dryrun=True):
 """ % nightly_age
     if dryrun:
         todelete = trans.execute("SELECT name FROM releases" + query).fetchall()
+        print "Releases rows to be deleted:"
         if todelete:
-            print "Releases rows to be deleted:"
             print "\n".join(itertools.chain(*todelete))
+        else:
+            print "  - None"
     else:
         trans.execute("DELETE releases FROM releases" + query)
 
 
 def cleanup_releases_history(trans, dryrun=True):
-    query = """
-WHERE name LIKE '%%%%latest'
-AND timestamp<1000*UNIX_TIMESTAMP(NOW()-INTERVAL 14 DAY);
-"""
+    num_to_delete = 100
+    find_records_latest = """
+        SELECT name, change_id
+        FROM releases_history
+        WHERE name LIKE '%%%%latest'
+          AND timestamp<1000*UNIX_TIMESTAMP(NOW()-INTERVAL 14 DAY)
+        ORDER BY change_id
+        LIMIT %d """ % num_to_delete
+
+    find_records_nightly = """
+        SELECT name, change_id
+        FROM releases_history
+        WHERE name LIKE '%%%%nightly%%%%'
+          AND name NOT LIKE '%%%%latest'
+          AND timestamp<1000*UNIX_TIMESTAMP(NOW()-INTERVAL 7 DAY)
+        ORDER BY change_id
+        LIMIT %d """ % num_to_delete
+
+    total_deleted = 0
+
     if dryrun:
-        todelete = trans.execute("SELECT name, change_id FROM releases_history" + query).fetchall()
+        todelete = trans.execute(find_records_latest).fetchall()
         if todelete:
-            print "Releases history rows to be deleted:"
+            print "Releases history (latest) rows to be deleted:"
             for key, group in itertools.groupby(todelete, lambda x: x[0]):
-                print "%s: %s history rows" % (key, len(list(group)))
+                print "  - %s: %s history rows" % (key, len(list(group)))
     else:
-        trans.execute("DELETE releases_history FROM releases_history" + query)
+        del_query = """
+            DELETE R.*
+            FROM releases_history R
+            WHERE R.change_id IN (   -- use a subquery to work around mysql limitation
+                SELECT X.change_id   -- of select'ing from the same table as a DML
+                FROM ( """ + find_records_latest + ") X)"
+
+        results = trans.execute(del_query)
+        if results:
+            print "Deleted %s 'latest' records" % results.rowcount
```

Comment:
It would be good to print the distinct names that were deleting history for. On the off chance we bugger this up in the future, or otherwise are trying to figure out howe some history disappeared, we'd be able to see the names in the output.